src/java/htsjdk/samtools/reference/IndexedFastaSequenceFile.java

```diff
@@ -202,10 +251,20 @@ public ReferenceSequence getSubsequenceAt( String contig, long start, long stop
             startOffset += Math.max((int)(startOffset%bytesPerLine - basesPerLine + 1),0);
 
             try {
-                 startOffset += channel.read(channelBuffer,indexEntry.getLocation()+startOffset);
+                if (channel instanceof FileChannel) {
+                    startOffset += ((FileChannel) channel).read(channelBuffer,indexEntry.getLocation()+startOffset);
```

Comment:
FileChannel takes advantage of native code, so I wanted to leave that code path unchanged. I've now split it out into a method as @lbergelson suggested.

@tfenne the semantics are the same for both, since after a positioned read FileChannel does not change the position; see http://docs.oracle.com/javase/7/docs/api/java/nio/channels/FileChannel.html#read(java.nio.ByteBuffer,%20long), where it says "This method does not modify this channel's position."